### DISCUSSION TITLE
Ability to run containerd without k3s

### DISCUSSION THREAD

**Discussion by retr0h** [k3s-io/k3s#8036] [General]
Is it possible to configure k3s such that only containerd starts up?

I'd like to pre-seed the local registry with container images, which would be pulled down from our remote registry, and added to the local containrd image registry.  However, I do not want to start k3s yet, as I need to ensure the images were successfully pulled first, prior to pivoting the system to a new k3s architecture.

**Comment by brandond** [comment]
No, there is not currently any way to do that.

There is however some undocumented functionality to pre-pull images during containerd startup. Similar to how you can place image tarballs in `/var/lib/rancher/k3s/agent/images` for preload on airgapped nodes, any .txt files found in that directory will be treated as an image list, and the images pulled before the kubelet is started.

**Reply by retr0h** [reply]
Yeah, understood.  For us, it is about accounting for unreliable networks and small links, prior to pivoting a system over to k3s.  We do not want to deploy k3s for the first time, if we do not first have the images downloaded.

**Reply by brandond** [reply]
That sounds like a good case for preloading airgap images and/or running a pull-through registry mirror. Alternately, you can install your Linux distribution's containerd package, configure and prepare it however you like, and point K3s at it with the `--container-runtime-endpoint` option.
